Controller.php

```diff
@@ -280,4 +281,23 @@ private function getPrettyDateForSite($datetime, $period, $timezone)
 
         return $prettyDate;
     }
+
+    private function fetchSites($alert)
+    {
+        $sites = SitesManagerApi::getInstance()->getSitesWithAtLeastViewAccess();
```

Comment:
As some users (and I think even clients) have like 50k websites it might be faster to call `getSitesIdWithAtLeastViewAccess` and then later fetch the names of the sites only for the sites actually needed. It might not make a big difference but could. Would require some testing with 50k websites and a regular user (not super user) whether it is still fast. In case you will switch back to the original site selector then this is not needed anymore anyway